python/ray/tune/trial_runner.py

```diff
@@ -396,88 +398,66 @@ def add_trial(self, trial):
+            format (str): Output format (see tablefmt in tabulate API).
+            max_rows (int): Maximum number of rows in the trial table.
+        """
         messages = ["== Status =="]
         messages.append(self._scheduler_alg.debug_string())
         messages.append(self.trial_executor.debug_string())
-        messages.append(self._memory_debug_string())
-        return messages
-
-    def _memory_debug_string(self):
-        try:
-            import psutil
-            total_gb = psutil.virtual_memory().total / (1024**3)
-            used_gb = total_gb - psutil.virtual_memory().available / (1024**3)
-            if used_gb > total_gb * 0.9:
-                warn = (": ***LOW MEMORY*** less than 10% of the memory on "
-                        "this node is available for use. This can cause "
-                        "unexpected crashes. Consider "
-                        "reducing the memory used by your application "
-                        "or reducing the Ray object store size by setting "
-                        "`object_store_memory` when calling `ray.init`.")
-            else:
-                warn = ""
-            return "Memory usage on this node: {}/{} GiB{}".format(
-                round(used_gb, 1), round(total_gb, 1), warn)
-        except ImportError:
-            return ("Unknown memory usage. Please run `pip install psutil` "
-                    "(or ray[debug]) to resolve)")
+        messages.append(memory_debug_string())
+
+        delim = "<br>" if format == "html" else "\n"
+        trials = self.get_trials()
+
+        if len(trials) < 1:
+            return delim.join(messages) + delim
+
+        num_trials = len(trials)
+        num_trials_per_state = collections.defaultdict(int)
+        for t in trials:
+            num_trials_per_state[t.status] += 1
+        messages.append("Number of trials: {} ({})"
+                        "".format(num_trials, num_trials_per_state))
+        if num_trials > max_rows:
+            messages.append("Table truncated to {} rows.".format(max_rows))
+
+        trial_table = []
+        if not parameters:
+            parameters = set().union(*[t.evaluated_params for t in trials])
+        metrics = list(metrics or DEFAULT_RESULT_KEYS)
+
+        for i in range(min(num_trials, max_rows)):
+            trial = trials[i]
+            result = flatten_dict(trial.last_result)
+            trial_info = [trial.trial_id, trial.status]
+            trial_info += [
+                result.get(CONFIG_PREFIX + param) for param in parameters
+            ]
+            trial_info += [result.get(metric) for metric in metrics]
+            trial_table.append(trial_info)
+
+        parsed_parameters = [param for param in parameters]
+        keys = ["Trial ID", "Status"] + parsed_parameters + metrics
+        messages.append(
+            tabulate(
+                trial_table, headers=keys, tablefmt=format, showindex=False))
+
+        return delim.join(messages) + delim
```

Comment:
do you think we can put this entire logging thing into a separate file? there's not much dependency on the trial runner.